### PR TITLE
Fix Swift code completion for Development Pods

### LINKED ISSUE #3777: Swift code completion not working on Development Pods
I posted my trouble on [Stack Overflow](http://stackoverflow.com/q/31091384/2203572), to which I was told by others experiencing this as well to post a bug report.

I have a project that uses various private internal pods, and upon installing them locally as Development Pods, those written in Swift have trouble code completing. Xcode complains of a "Symbol Not Found" when command clicking, even on `import Foundation`. Everything complies and runs fine with no errors.

As soon as I install them from remote and not as Development Pods however, everything code completes with no issues.

I have tried various solutions, namely:
* deleting Xcode's derived data
* selecting a different simulator device to force a re-index
* deleting Cocoapod's cache (both in the library and the Podfile.lock and Pods directory)
* running pod deintegrate and then pod install
* restarting Xcode and computer
* using different versions of Xcode (6.3.2, 6.4, 7.0-beta2)
* on a completely different machine, with a brand new project and installing Alamofire as a development pod

Any help would be greatly appreciated!

### PR DESCRIPTION
**Pull request by segiddins**
Closes #3777.

- [x] CHANGELOG
- [x] Tests
  - [x] Add unit test
  - [x] Fix existing specs

### CODE REVIEW

**Comment by segiddins** [comment]
@neonichu ping on this?

**Comment by segiddins** [comment]
Actually, I'm not sure we can safely do this, as won't this make all the added paths absolute instead of relative? Can you double check?

**Comment by neonichu** [comment]
Will do.

**Comment by segiddins** [comment]
Scratch my previous comment, this should work

**Comment by segiddins** [comment]
@neonichu ping on this?

**Comment by segiddins** [comment]
:+1:

File: lib/cocoapods/project.rb [hunk 1]
```diff
@@ -179,13 +179,13 @@ def pod_support_files_group(pod_name, dir)
     # @return [PBXFileReference] The new file reference.
     #
     def add_file_reference(absolute_path, group, reflect_file_system_structure = false)
-      file_path_name = Pathname.new(absolute_path)
+      file_path_name = absolute_path.is_a?(Pathname) ? absolute_path : Pathname.new(absolute_path)
```
Comment (segiddins) [inline lib/cocoapods/project.rb]: why is this change necessary?

Comment (neonichu) [inline lib/cocoapods/project.rb]: It's not really, but it made stubbing things on a specific `Pathname` instance possible :)

Comment (segiddins) [inline lib/cocoapods/project.rb]: gotcha :+1:

**Pull request merged**